cmd/kubeless/function.go

```diff
@@ -100,14 +101,81 @@ func parseMemory(mem string) (resource.Quantity, error) {
 	return quantity, nil
 }
 
-func getFunctionDescription(funcName, ns, handler, funcContent, deps, runtime, topic, schedule, runtimeImage, mem string, triggerHTTP bool, envs, labels []string, defaultFunction spec.Function) (f *spec.Function, err error) {
+func getFileSha256(file string) (string, error) {
+	h := sha256.New()
+	ff, err := os.Open(file)
+	if err != nil {
+		return "", err
+	}
+	defer ff.Close()
+	_, err = io.Copy(h, ff)
+	if err != nil {
+		return "", err
+	}
+	checksum := hex.EncodeToString(h.Sum(nil))
+	return checksum, err
+}
+
+func uploadFunction(file string, cli kubernetes.Interface) (string, string, string, error) {
+	stats, err := os.Stat(file)
+	if err != nil {
+		return "", "", "", err
+	}
+	if stats.Size() > int64(50*1024*1024) { // TODO: Make the max file size (50 MB) configurable
+		err = errors.New("The maximum size of a function is 50MB")
+		return "", "", "", err
+	}
+	// If an object storage service is not available check
+	// that the file is not over 1MB to store it as a Custom Resource
+	if stats.Size() > int64(1*1024*1024) {
+		err = errors.New("Unable to deploy functions over 1MB withouth a storage service")
+		return "", "", "", err
+	}
+	functionBytes, err := ioutil.ReadFile(file)
+	if err != nil {
+		return "", "", "", err
+	}
+	if err != nil {
+		return "", "", "", err
+	}
+	fileType := http.DetectContentType(functionBytes)
+	var function, contentType string
+	if strings.Contains(fileType, "text/plain") {
+		function = string(functionBytes)
+		contentType = "text"
+	} else {
+		function = base64.StdEncoding.EncodeToString(functionBytes)
+		contentType = "base64"
+		if strings.Contains(fileType, "zip") {
```

Comment:
It's better use filepath.Ext() to detect the extension. Otherwise, if the file contains `zip` in somewhere else that would lead to a wrong detection.